target_resci/__init__.py

```diff
@@ -141,93 +129,100 @@ def headers(self):
                           giveup=singer.utils.exception_is_4xx,
                           max_tries=8,
                           on_backoff=_log_backoff)
-    def send(self, file_type, file_name):
-        """Send the given data to ReSci, retrying on exceptions"""
+    def send_files(self):
+        """Sends all files to ReSci, retrying on exceptions"""
         url = self.resci_url
         headers = self.headers()
         ssl_verify = True
         if os.environ.get("TARGET_RESCI_SSL_VERIFY") == 'false':
             ssl_verify = False
 
-        files = {file_type: open(file_name, 'rb')}
+        files_to_send = {}
+        for stream, file in self.stream_files.items():
+            files_to_send[stream] = open(file.name, 'rb')
+
         params = {'import_type': self.import_type}
         response = self.session.post(url,
                                      data=params,
                                      headers=headers,
-                                     files=files,
+                                     files=files_to_send,
                                      verify=ssl_verify)
 
         response.raise_for_status()
         return response
 
-    def flatten(self, d, parent_key='', sep='__'):
-        """Flattens dictionary"""
-        items = []
-        for k, v in d.items():
-            new_key = parent_key + sep + k if parent_key else k
-            if isinstance(v, MutableMapping):
-                items.extend(self.flatten(v, new_key, sep=sep).items())
-            else:
-                items.append((new_key, str(v) if type(v) is list else v))
-        return dict(items)
-
-    def create_file(self, messages, file_type, batch_count):
-        """Creates a file and writes JSON"""
-        filename = '{}-{}-{:03}.json'.format(file_type, self.file_id, batch_count)
-        LOGGER.debug('Filename: %s', filename)
-        with open(filename, "w") as outfile:
-            for msg in messages:
-                flattened_record = self.flatten(msg.record)
-                json.dump(flattened_record, outfile)
-                outfile.write("\n")
-        return filename
-
-    def handle_batch(self, messages, batch_count, dry_run):
-        """Handle messages by sending them to ReSci as import file.
+    def get_or_create_file(self, stream):
+        """Get the opened file for a stream, creating the file if it doesn't exist"""
+        file = self.stream_files.get(stream)
+
+        if file is None:
+            filename = '{}-{}.json'.format(stream, self.file_id)
+            LOGGER.debug('Opening file %s', filename)
+            file = open(filename, 'w')
+            self.stream_files[stream] = file
+
+        return file
+
+    def write_to_file(self, messages, stream, batch_count):
```

Comment:
essentially instead of opening a file for each batch, this just looks up the correct file from the `stream_files` dict and appends to that